integration/entity.go

```diff
@@ -67,18 +70,16 @@ func newEntity(
 		lock:        &sync.Mutex{},
 	}
 
-	mapAttr := make([]map[string]string, len(idAttr))
-	for i, attr := range idAttr {
-		mapAttr[i] = map[string]string{
-			attr.key: attr.value,
-		}
+	idAttrKVs := []AttrKV{}
+	for _, a := range idAttrs {
+		idAttrKVs = append(idAttrKVs, a.KV())
```

Comment:
I prefer avoiding append here since we know the total length of the new slice. Using `make(x, len(idAttrKVs)` and then access by index we avoid new allocation in the slice.